pom.xml

```diff
@@ -249,7 +249,7 @@
     <dependency>
       <groupId>org.apache.sling</groupId>
       <artifactId>org.apache.sling.testing.paxexam</artifactId>
-      <version>3.1.0</version>
+      <version>4.0.0</version>
```

Comment:
This seems to be the culprit of the failing test.

Reply:
I added this for Java 17 support, see https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=255071690#CommonIssueswhenbuildingwithnewerJDKs(11or17)-Pax-Examtestfailures

Reply:
This update brought `org.apache.felix.jetty-5.0.0`, which uses Jetty 11. Unfortunately, there's a Servlet Request Wrapper in Felix that doesn't correctly return the set attributes if they were `RequestDispatcher` attributes - see https://github.com/apache/felix-dev/blob/d691a59c9e78749fad20a10a002c969e4e583800/http/wrappers/src/main/java/org/apache/felix/http/javaxwrappers/ServletRequestWrapper.java#L108-L182

@cziegeler will fix this in Felix and for this test setup we could override the Jetty version.

Reply:
Nice catch!